### PR TITLE
feat(store): integrate `store-rocksdb` submodule

### PR DESCRIPTION
**Pull request by Pengzna**
Subtask of https://github.com/apache/incubator-hugegraph/issues/2265 and https://github.com/apache/incubator-hugegraph/issues/2482

The corresponding tests will be merged after `store-core` and `store-node`

Detailed doc can be seen in [this link](https://hugegraph.feishu.cn/wiki/M8Sdw2s0YiP2YCkQ9zxcO2vVnth)

### CODE REVIEW

File: hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/util/ZipUtils.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public static void decompress(final String sourceFile, final String outputDir,
+                                  final Checksum checksum) throws IOException {
+        try (final FileInputStream fis = new FileInputStream(sourceFile);
+             final CheckedInputStream cis = new CheckedInputStream(fis, checksum);
+             final ZipInputStream zis = new ZipInputStream(new BufferedInputStream(cis))) {
+            ZipEntry entry;
+            while ((entry = zis.getNextEntry()) != null) {
+                final String fileName = entry.getName();
```
[the remark this answers is not in the document]
Comment (Pengzna) [reply hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/util/ZipUtils.java:80]: this is a Zip Slip sec issue, fixed

File: hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/SessionOperatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        final byte BYTE_MAX_VALUE = (byte) 0xff;
+        assert bytes.length > 0;
+        byte last = bytes[bytes.length - 1];
+        if (last != BYTE_MAX_VALUE) {
+            bytes[bytes.length - 1] += 0x01;
+        } else {
+            // Process overflow (like [1, 255] => [2, 0])
+            int i = bytes.length - 1;
+            for (; i > 0 && bytes[i] == BYTE_MAX_VALUE; --i) {
+                bytes[i] += 0x01;
```
[the remark this answers is not in the document]
Comment (Pengzna) [reply hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/SessionOperatorImpl.java]: fixed

File: hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/SessionOperatorImpl.java [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+            }
+            if (bytes[i] == BYTE_MAX_VALUE) {
+                assert i == 0;
+                throw new DBStoreException("Unable to increase bytes: %s", Bytes.toHex(bytes));
+            }
+            bytes[i] += 0x01;
```
[the remark this answers is not in the document]
Comment (Pengzna) [reply hugegraph-store/hg-store-rocksdb/src/main/java/org/apache/hugegraph/rocksdb/access/SessionOperatorImpl.java]: fixed

File: hugegraph-store/hg-store-rocksdb/src/main/java/place-holder.txt
Comment (imbajin) [inline hugegraph-store/hg-store-rocksdb/src/main/java/place-holder.txt:1]: why we need this empty file?

File: hugegraph-store/hg-store-rocksdb/pom.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>hg-store-rocksdb</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.hugegraph</groupId>
+            <artifactId>hugegraph-common</artifactId>
+            <version>1.2.0</version>
```
Comment (imbajin) [inline hugegraph-store/hg-store-rocksdb/pom.xml:37]:
```suggestion
            <version>1.3.0</version>
```

TODO: also we should use the param to unify it

File: hugegraph-store/hg-store-rocksdb/src/main/java/place-holder.txt
Comment (Pengzna) [reply hugegraph-store/hg-store-rocksdb/src/main/java/place-holder.txt:1]: I don’t know what the use of this file is. Ignore it for now and wait for confirmation before processing it.

**Review by imbajin — APPROVED** [review]

**Review by VGalaxies — APPROVED** [review]

**Pull request merged**